lib/core/include/irods/connection_pool.hpp

```diff
@@ -15,6 +16,33 @@
 
 namespace irods
 {
+    /// Defines various options for changing the behavior of a connection pool.
+    ///
+    /// \since 4.3.1
+    struct connection_pool_options
+    {
+        /// Defines the number of API requests that can be sent via a connection before
+        /// refreshing it.
+        ///
+        /// Independent of any other option.
+        ///
+        /// \since 4.3.1
+        std::optional<std::int16_t> refresh_connections_after_n_retrievals;
```

Comment:
This name kind of reads like a boolean. I may change it.

`max_retrievals_before_refresh`

hmm ...

Reply:
`number_of_retrievals_before_refresh`

Reply:
`retrieval_quantity_before_refresh`?

Reply:
Went with `number_of_retrievals_before_connection_refresh`.